### ISSUE TITLE
oauth2server: confidential vs public clients

### ISSUE THREAD

**Issue by lnielsen** [inveniosoftware/invenio#2113]
- [x] New application form should ask user for client type (confidential/public).
- [x] Refresh tokens MUST NOT be issued to public clients.
- [x] Refresh tokens SHOULD be issued to confidential clients.
- [x] Add warning that public clients must validate the redirect URI.
- [ ] Add tests

Depends on #2112

**Comment by jirikuncar** [comment]
ping @lnielsen with https://github.com/inveniosoftware/invenio/issues/2113#issuecomment-59894527

**Comment by lnielsen** [comment]
Let's close it? It's been in production long and I haven't discovered any issues. If I discover any problems, I'll open a new ticket and add required tests.
